api/src/main/resources/workbench.yaml

```diff
@@ -967,6 +995,59 @@ definitions:
         format: int64
         description: Milliseconds since the UNIX epoch.
 
+  MaterializeCohortRequest:
+    type: object
+    properties:
+      cohortName:
+        description: >
+         The name of a cohort that is to be evaluated. Either this or cohortSpec should be specified
+        type: string
+      cohortSpec:
```

Comment:
I've been pondering that. @freemabd1 

Advantages to just passing around arbitrary JSON:

- We can support different cohort JSON schemas (say for different cohort builders or versions of it) with a single set of endpoints that deal with cohorts (allows for a plugin architecture essentially)
- Researchers could throw in their own random JSON attributes if they wanted to (say for documentation purposes).

Disadvantages:

- Easier to screw things up on the client and pass in invalid JSON
- Not self-documenting
- If we want to provide an API for constructing a cohort spec programatically, we have to ship a separate library for that
- We still have to update the server whenever we change the supported formats, so it's not truly decoupled

Anyway, this is how things are for now, but I'm open to debate on the subject. :)

Reply:
Seems like it would be very little overhead to support multiple/future cohort schemas in this API while still documenting them here, e.g. one wrapper object to act as a union. Presumably this materialization API is meant to be called from notebooks, so it would be nice to standardize the API rather than making assumptions about this JSON blob in both places (here and CreateCohort from the cohort builder).

Would we actually want to allow arbitrary JSON here? Presumably we'd run into problems if we changed our JSON schema and it conflicted with a key a user stuck in there and you also run into bugs if the request is not validated, e.g. we might intepret a typo in the input JSON object as a user-defined key. Would be cleaner in my opinion to just attach these kinds of user labels explicitly on something like a `Cohort.labels` object key-value field.

Not part of this PR but would definitely strongly recommend going that route.

Reply:
Yeah, I could see going that route; it's a bigger discussion. 

I definitely think we want cohorts stored as JSON in the database; we don't want to have to update our DB schema every time the cohort spec changes, since it's going to change a lot. But we could still have an API that explicitly models cohorts. (Assuming that Swagger codegen is capable of generating objects that we're happy with.)

Anyway, let's chat with @freemabd1 and @jscherdin  about it (once Jon gets back in town?)

Reply:
sure.